src/google/google-calendars/googlecalendarsyncadaptor.cpp

```diff
@@ -588,37 +589,43 @@ void GoogleCalendarSyncAdaptor::updateLocalCalendarNotebooks(int accountId, cons
         }
     }
 
-    // Finally, request the events for each calendar.
-    // If the last sync was successful, we can do a fast sync (using change deltas).
-    // NOTE: this "since" value was written at the completion of the previous
-    // sync operation.  This means that there is a time-delta between when the
-    // sync occurs, and when the timestamp is written to the database.  Any
-    // local or remote modifications which occur during this time-delta might not
-    // be synced correctly... XXX TODO: FIXME!!!
-    QDateTime since = needCleanSync
-                    ? QDateTime()
-                    : lastSyncTimestamp(QLatin1String("google"),
```

Comment:
the lastSyncTimestamp function, IIRC, looks at a sync anchors database table (which is completely separate to the libsocialcache database).  I wonder if we can actually remove that Sync Anchors database altogether (would save some I/O etc).  I think we need to look carefully at the other sync adaptors to see if I made a similar mistake elsewhere - if so, we either need to: 1) use the sync anchors database instead of removing it, but store the the server-supplied timestamps instead of the local ones, 2) remove the sync anchors database and store the server-supplied timestamps in band with the synced data (eg, in the contacts database, etc).